core/src/main/java/tech/pegasys/eth2signer/core/signing/EthSecpArtifactSigner.java

```diff
@@ -36,4 +47,28 @@ public String getIdentifier() {
   public SecpArtifactSignature sign(final Bytes message) {
     return new SecpArtifactSignature(signer.sign(message.toArray()));
   }
+
+  @Override
+  public boolean verify(final Bytes message, final ArtifactSignature artifactSignature) {
+    checkArgument(artifactSignature instanceof SecpArtifactSignature);
+    final SecpArtifactSignature secpArtifactSignature = (SecpArtifactSignature) artifactSignature;
+    final Signature signature = secpArtifactSignature.getSignatureData();
+    final SignatureData signatureData =
+        new SignatureData(
+            Numeric.toBytesPadded(signature.getV(), 1),
+            Numeric.toBytesPadded(signature.getR(), 32),
+            Numeric.toBytesPadded(signature.getS(), 32));
+
+    final byte[] digest = Hash.sha3(message.toArrayUnsafe());
+
+    try {
+      final BigInteger signaturePublicKey = Sign.signedMessageHashToKey(digest, signatureData);
+      final BigInteger expectedPublicKey =
+          Numeric.toBigInt(EthPublicKeyUtils.toByteArray(signer.getPublicKey()));
+      return signaturePublicKey.equals(expectedPublicKey);
+    } catch (SignatureException e) {
+      LOG.error("Invalid signature", e);
```

Comment:
what would be the frequency of verify throwing signatureexception? if it is too high, this will probably produce too much noise. Also, `final`!!

Reply:
Wouldn't expect this happen that often